### PR TITLE
ENH Update status flags generally, not just for SiteTree

### PR DESCRIPTION
**Pull request by GuySartorelli**
https://github.com/silverstripe/silverstripe-framework/pull/11460 makes the `updateStatusFlags()` extension hook available for all models. In combination with other PRs linked in the below issue, status flag rendering has now been standardised.

This PR updates fluent to add fluent status flags to any fluent model, not just `SiteTree`, and removes custom code that renders flags in places like `LeftAndMain` breadcrumbs, gridfield edit form breadcrumbs, and gridfield rows. Those are all handled in core now.

Please note that this PR relies on https://github.com/silverstripe/silverstripe-framework/pull/11460 along with others.
**_Please do not merge this PR until all dependent PRs have been merged._**

Looks like this in CMSMain:
Screenshot from 2024-11-20 14-56-20

Looks like this in gridfield (yes that is a gridfield holding pages, no it is not CMSMain):
image
image

## Issue
- https://github.com/silverstripe/silverstripe-cms/issues/2947

### CODE REVIEW

File: src/Extension/FluentSiteTreeExtension.php [hunk 1]
```diff
@@ -158,33 +158,6 @@ protected function updateLink(&$link, &$action, &$relativeLink)
...
     }
 
-    /**
-     * Check whether the current page is exists in the current locale.
-     *
-     * If it is invisible then we add a class to show it slightly greyed out in the site tree.
-     *
-     * @param array $flags
-     */
-    protected function updateStatusFlags(&$flags)
```
Comment (ChloeSartorelli) [inline src/Extension/FluentSiteTreeExtension.php:168]: Nothing here is specific to SiteTree anymore. This is now split up between `FluentExtension` and `FluentVersionedExtension` along with the methods called from here.

File: src/Extension/FluentExtension.php [hunk 2]
```diff
@@ -1286,6 +1287,114 @@ protected function findRecordInLocale($locale, $table, $id)
...
+            if ($key === 'fluent' || str_starts_with($key, 'fluent ')) {
+                unset($flags[$key]);
+            }
+        }
+    }
+
+    /**
+     * Update status flags based on whether the current record is exists in the current locale.
+     */
+    protected function updateStatusFlags(array &$flags): void
```
Comment (ChloeSartorelli) [inline src/Extension/FluentExtension.php:1305]: See `FluentSiteTreeExtension` for where this code originated.

File: src/Extension/FluentVersionedExtension.php [hunk 3]
```diff
@@ -586,6 +586,76 @@ public static function prepoulateIdsInLocale($locale, $dataObjectClass, $populat
...
     }
 
+    /**
+     * Check whether the current record is exists in the current locale.
+     *
+     * If it is invisible then we add a class to show it slightly greyed out in the site tree.
+     *
+     * @param array $flags
+     */
+    protected function updateStatusFlags(array &$flags): void
```
Comment (ChloeSartorelli) [inline src/Extension/FluentVersionedExtension.php:596]: See `FluentSiteTreeExtension` for where this code originated.

File: src/Extension/FluentGridFieldExtension.php [hunk 4]
```diff
@@ -23,19 +21,6 @@
...
     use FluentAdminTrait;
-    use FluentBadgeTrait;
-
-    /**
-     * Push a badge to indicate the language that owns the current item
-     *
-     * @param DBField|null $badgeField
-     * @see VersionedGridFieldItemRequest::Breadcrumbs()
-     */
-    protected function updateBadge(&$badgeField)
```
Comment (ChloeSartorelli) [inline src/Extension/FluentGridFieldExtension.php:34]: That hook doesn't exist anymore - this is now handled through generalised status flag rendering in core.

File: src/Extension/FluentLeftAndMainExtension.php [hunk 5]
```diff
@@ -7,45 +7,23 @@
...
     {
         Requirements::javascript("tractorcow/silverstripe-fluent:client/dist/js/fluent.js");
         Requirements::css("tractorcow/silverstripe-fluent:client/dist/styles/fluent.css");
     }
 
-    /**
-     * @param ArrayList $breadcrumbs
-     * @see CMSMain::Breadcrumbs()
-     */
-    protected function updateBreadcrumbs(ArrayList $breadcrumbs)
```
Comment (ChloeSartorelli) [inline src/Extension/FluentLeftAndMainExtension.php:34]: This is now handled through generalised status flag rendering in core.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
I've tried rerunning tests after merging everything else, there are CI failures

**Comment by GuySartorelli** [comment]
Updated status flag tests and added a little more coverage than was already there.

**Review by emteknetnz — APPROVED** [review]
Seems to do what it's supposed to do

**Pull request merged**